modules/evaluator/main.go

```diff
@@ -33,28 +36,45 @@ func New(
 	model string,
 ) *Evaluator {
 	return &Evaluator{
-		Docs:          docs,
-		InitialPrompt: initialPrompt,
-		WriterModel:   model,
+		Docs:           docs,
+		InitialPrompt:  initialPrompt,
+		EvaluatorModel: model,
 	}
 }
 
-func (m *Evaluator) llm() *dagger.LLM {
-	return dag.LLM(dagger.LLMOpts{Model: m.WriterModel})
+type Eval interface {
+	Name(context.Context) (string, error)
+	Prompt(base *dagger.LLM) *dagger.LLM
+	Check(ctx context.Context, prompt *dagger.LLM) error
+
+	DaggerObject
 }
 
-func (m *Evaluator) env() *dagger.Env {
-	env := dag.Env().
-		WithWorkspaceInput("workspace", m.work(), "A space for you to work in.")
-	if m.Docs != nil {
-		env = env.WithFileInput("docs", m.Docs,
-			"The documentation the model is meant to adhere to.")
+func (m *Evaluator) WithEval(ctx context.Context, eval Eval) (*Evaluator, error) {
+	id, err := eval.(interface {
+		ID(context.Context) (EvalID, error)
+	}).ID(ctx)
+	if err != nil {
+		return nil, err
 	}
-	if m.InitialPrompt != nil {
-		env = env.WithFileInput("initialSystemPrompt", m.InitialPrompt,
-			"An initial system prompt to evaluate and improve.")
+	// FIXME: it would be nice to not have to do this workaround. it's hard
+	// because we want to accept Eval, but then the type has no AsWorkspaceEval().
+	//
+	// fortunately the IDs are the same nonetheless, so we can just convert it
+	// with the available plumbing
+	m.Evals = append(m.Evals, dag.LoadWorkspaceEvalFromID(dagger.WorkspaceEvalID(id)))
```

Comment:
Ran into some friction with interfaces here, but the workaround ain't too bad. cc @sipsma 

The problem is kind of tricky: there's an `Eval` interface in the `evaluator/workspace/` module, and it's the same interface defined locally here, but I have to convert it from my interface to the `WorkspaceEval` version.